File: 0035-search-insert-position/0035-search-insert-position.cpp
#include <vector>

class Solution {
public:
    int searchInsert(vector<int>& a, int target) {
        int n = a.size();
        vector<int> p;

        for (int i = 0; i < n; i++) {
            if (a[i] < target) {
                p.push_back(a[i]);
            } else {
                // If the current element is greater than or equal to the target,
                // the correct position to insert the target is found.
                return i;
            }
        }

        // If we reach here, it means the target is greater than all elements in the vector.
        // Therefore, insert it at the end.
        return n;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] a, int target) {
        int n=a.length;
        int low=0,high=n-1;
        while(low<=high){
            int mid=high+(high-low)/2;
            if(a[mid]==target){
                return mid;
            }
            else if(a[mid]>target){
            high=mid-1;
            }
            else{
                low=mid+1;
            }
        }
        return low;
        
    }
}